drivers/flash/soc_flash_stm32f4.c

```diff
@@ -255,6 +254,12 @@ static struct flash_driver_api flash_stm32f4_api = {
 static int stm32f4_flash_init(struct device *dev)
 {
 	struct flash_priv *p = dev->driver_data;
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(sector); i++) {
+		sector[i].start = CONFIG_FLASH_BASE_ADDRESS +
+			i * sector[i - 1].len;
+	}
```

Comment:
The issue is not the holes problem, but is that the base address can be any address in the flash. Like for the dual bank scenario we have, the bootloader would start at 0x0, the first partition at 0x20000. If the flash base address is actually 0x20000, then how are you going to be able to erase sector 0x0?